superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/AdhocMetricEditPopover.test.tsx

```diff
@@ -0,0 +1,184 @@
+  columns: [
+    {
+      id: 1342,
+      column_name: 'highest_degree_earned',
+      expression:
+        "CASE \n  WHEN school_degree = 'no high school (secondary school)' THEN 'A. No high school (secondary school)'\n  WHEN school_degree =  'some high school' THEN 'B. Some high school'\n  WHEN school_degree = 'high school diploma or equivalent (GED)' THEN 'C. High school diploma or equivalent (GED)'\n  WHEN school_degree = 'associate''s degree' THEN 'D. Associate''s degree'\n  WHEN school_degree = 'some college credit, no degree' THEN 'E. Some college credit, no degree'\n  WHEN school_degree = 'bachelor''s degree' THEN 'F. Bachelor''s degree'\n  WHEN school_degree = 'trade, technical, or vocational training' THEN 'G. Trade, technical, or vocational training'\n  WHEN school_degree = 'master''s degree (non-professional)' THEN 'H. Master''s degree (non-professional)'\n  WHEN school_degree = 'Ph.D.' THEN 'I. Ph.D.'\n  WHEN school_degree = 'professional degree (MBA, MD, JD, etc.)' THEN 'J. Professional degree (MBA, MD, JD, etc.)'\nEND",
+      type: 'STRING',
+    },
+  ],
+});
+
+test('Should render', () => {
+  const props = createProps();
+  render(<AdhocMetricEditPopover {...props} />);
+  expect(screen.getByTestId('metrics-edit-popover')).toBeVisible();
+});
+
+test('Should render correct elements', () => {
+  const props = createProps();
+  render(<AdhocMetricEditPopover {...props} />);
+
+  expect(screen.getByRole('tablist')).toBeVisible();
+
+  expect(screen.getByRole('tab', { name: 'Custom SQL' })).toBeVisible();
+  expect(screen.getByRole('tab', { name: 'Simple' })).toBeVisible();
+  expect(screen.getByRole('tab', { name: 'Saved' })).toBeVisible();
+
+  expect(screen.getByRole('tabpanel', { name: 'Saved' })).toBeVisible();
+
+  expect(screen.getByRole('button', { name: 'Resize' })).toBeVisible();
+  expect(screen.getByRole('button', { name: 'Save' })).toBeVisible();
+  expect(screen.getByRole('button', { name: 'Close' })).toBeVisible();
+});
+
+test('When click on "Close" should call onClose', () => {
+  const props = createProps();
+  render(<AdhocMetricEditPopover {...props} />);
+  expect(props.onClose).toBeCalledTimes(0);
+  userEvent.click(screen.getByRole('button', { name: 'Close' }));
+  expect(props.onClose).toBeCalledTimes(1);
+});
+
+test('When click on "Save" should call onChange and onClose', () => {
+  const props = createProps();
+  render(<AdhocMetricEditPopover {...props} />);
+  expect(props.onChange).toBeCalledTimes(0);
+  expect(props.onClose).toBeCalledTimes(0);
+  userEvent.click(screen.getByRole('button', { name: 'Save' }));
+  expect(props.onChange).toBeCalledTimes(1);
+  expect(props.onChange).toBeCalledWith(
+    {
+      id: 64,
+      metric_name: 'count',
+      expression: 'COUNT(*)',
+    },
+    {
+      id: 64,
+      metric_name: 'count',
+      expression: 'COUNT(*)',
+    },
+  );
+  expect(props.onClose).toBeCalledTimes(1);
+});
+
+test('Should switch to tab:Simple', () => {
```

Comment:
```suggestion
test('Should switch to "Simple" tab', () => {
```

Reply:
In these cases I prefer to keep it this way because `Simple ` is the value of input `tab`. When I'm writing tests I try to highlight in the title for which value is which attribute.